### PR TITLE
feat(alloydbomni): add service and dependant resources

### PR DESCRIPTION
**Pull request by byashimov**
Add `alloydbomni`, `aiven_alloydbomni_user`, `aiven_alloydbomni_database` BETA resources and datasources

### CODE REVIEW

File: internal/sdkprovider/service/alloydbomni/alloydbomni_user.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		Sensitive:        true,
+		Computed:         true,
+		DiffSuppressFunc: schemautil.EmptyObjectDiffSuppressFunc,
+		Description:      "The password of the service user.",
+	},
+	"alloydbomni_allow_replication": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Allows replication. For the default avnadmin user this attribute is required and is always `true`.",
+	},
```
[the remark this answers is not in the document]
Comment (byashimov) [reply internal/sdkprovider/service/alloydbomni/alloydbomni_user.go:38]: Renamed to `pg_allow_replication`

File: internal/sdkprovider/service/alloydbomni/alloydbomni_test.go
Comment (rriski) [inline internal/sdkprovider/service/alloydbomni/alloydbomni_test.go:1]: There are many tests in this file that mirror the pg tests. I think we can remove many of them and keep the most. I would go as far as only keeping `TestAccAivenServiceAlloyDBOmni_service_account_credentials`, what do you think?

File: docs/data-sources/alloydbomni.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+- `project_vpc_id` (String) Specifies the VPC the service should run in. If the value is not set the service is not run inside a VPC. When set, the value should be given as a reference to set up dependencies correctly and the VPC must be in the same cloud and region as the service itself. Project can be freely moved to and from VPC after creation but doing so triggers migration to new servers so the operation can take significant amount of time to complete if the service has a lot of data.
+- `service_account_credentials` (String, Sensitive) Google Service Account Credentials
```
Comment (rriski) [inline docs/data-sources/alloydbomni.md:45]: Can we somehow provide more information what the expected format of the string is? APIs will fail if the structure is incorrect.

File: internal/sdkprovider/service/alloydbomni/alloydbomni_test.go
Comment (rriski) [inline internal/sdkprovider/service/alloydbomni/alloydbomni_test.go:1]: Can we add a test with invalid service account config and validate that its not accepted by the API (or better yet, not accepted by tf)?

File: internal/sdkprovider/service/alloydbomni/alloydbomni_user_test.go
Comment (rriski) [inline internal/sdkprovider/service/alloydbomni/alloydbomni_user_test.go:1]: These tests are mirrored from pg. As they use the same APIs underneath, maybe we can make this suite a bit shorter? Now these tests are essentially duplicate. Maybe keep just `TestAccAivenPGUser_basic`?

File: internal/sdkprovider/userconfig/service/service.go [hunk 3]
```diff
@@ -48,6 +50,13 @@ func GetUserConfig(kind string) *schema.Schema {
 // GetFieldMapping returns TF fields to Json fields mapping (in unix-path way)
 func GetFieldMapping(kind string) map[string]string {
 	return map[string]map[string]string{
+		"alloydbomni": {
+			"ip_filter_object":                  "ip_filter",
+			"ip_filter_string":                  "ip_filter",
+			"pg/pg_partman_bgw__dot__interval":  "pg/pg_partman_bgw.interval",
+			"pg/pg_partman_bgw__dot__role":      "pg/pg_partman_bgw.role",
+			"pg/pg_stat_statements__dot__track": "pg/pg_stat_statements.track",
+		},
```
Comment (rriski) [inline internal/sdkprovider/userconfig/service/service.go:59]: Fields names with dots in them make me sad 😞

File: internal/sdkprovider/service/alloydbomni/alloydbomni_user_test.go
Comment (byashimov) [reply internal/sdkprovider/service/alloydbomni/alloydbomni_user_test.go:1]: Hey. We have addressed several backend bugs, thanks to these tests. I would keep them.

File: internal/sdkprovider/service/alloydbomni/alloydbomni_test.go
Comment (byashimov) [reply internal/sdkprovider/service/alloydbomni/alloydbomni_test.go:1]: Added schema validation.

**Review by rriski — APPROVED** [review]

**Pull request merged**